packages/core/src/app/customer/CheckoutButtonContainer.tsx

```diff
@@ -27,22 +37,63 @@ const CheckoutButtonContainer: FunctionComponent<CheckoutButtonsListOnTopProps &
         },
     } = checkoutState;
     const config = getConfig();
+    const methodIds = config?.checkoutSettings.remoteCheckoutProviders ?? [];
+
+    useEffect(() => {
+        const changeState = setTimeout(()=>{
+            if(!isInitializingCustomer()) {
+                setIsLoading(false);
```

Comment:
What does this `useEffect` block do?

Reply:
`isInitializingCustomer()` toggles between `true` and `false` during loading, forcing the loading skeleton to render whenever it is `false`, therefore I use this block to hide the skeleon when the loading is actully done.

I also tried other state selectors such as `isPending`, `isLoadingMethod`. They all act the same way.

Reply:
Do we know the reason why it toggles between the values? Is that because it processes requests for each button/method id?

Reply:
Right, it sounds like we're missing something like `isInitialized` (https://github.com/bigcommerce/checkout-sdk-js/blob/e9d7830aae32d48fbaf825c232a7a99b0099767d/packages/core/src/checkout-buttons/checkout-button-selector.ts#L12). We have it for checkout buttons, but not for customer step. Could we consider adding one instead of using a hardcoded timeout? As far as I can tell, the timeout solution won't work if the initialisation takes longer than 100ms. We need to actually set a flag somewhere once the initialisation is complete.

Reply:
I think that's a reasonable approach, and makes the overall solution more robust than relying on timeouts cc @bc-peng